File: 5 kyu/Maximum subarray sum/main.cpp
#include <vector>
#include <algorithm>

int maxSequence(const std::vector<int>& arr){
  if(arr.size() == 0){return 0;}
  int maxsum = arr[0], current_sum = arr[0];
  for(int it = 1;it < (int)arr.size(); ++it)
  {
    current_sum = std::max(arr[it], current_sum + arr[it]);
    maxsum = std::max(maxsum, current_sum);
  }
  if (maxsum <= 0){return 0;}
  return maxsum;
}

File: 5 kyu/Maximum subarray sum/main.py
def max_sequence(arr):
    if(len(arr)== 0):
        return 0
    maxsum, current_sum = arr[0],arr[0]
    for it in range(1,len(arr)):
        current_sum = max(arr[it],current_sum + arr[it])
        maxsum = max(maxsum,current_sum)
    if(maxsum <= 0):
        return 0
    return maxsum
